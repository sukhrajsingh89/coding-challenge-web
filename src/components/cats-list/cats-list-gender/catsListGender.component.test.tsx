import React from "react";
import { render, cleanup,screen } from "@testing-library/react";
import CatsListGender from "./catsListGender.component";

describe("<CatsList", () => {
  afterEach(() => {
    cleanup();
  });

  const renderComponent = (props = {}) => {
    return render(<CatsListGender userObject={[]} gender={undefined} />);
  };

  it("it should render the component", () => {
    const props = {
      userObject: [
        {
          name: "Bob",
          gender: "Male",
          age: 23,
          pets: [
            { name: "Garfield", type: "Cat" },
            { name: "Fido", type: "Dog" },
          ],
        },
        {
          name: "Jennifer",
          gender: "Female",
          age: 18,
          pets: [{ name: "Garfield", type: "Cat" }],
        },
        { name: "Steve", gender: "Male", age: 45, pets: null },
        {
          name: "Fred",
          gender: "Male",
          age: 40,
          pets: [
            { name: "Tom", type: "Cat" },
            { name: "Max", type: "Cat" },
            { name: "Sam", type: "Dog" },
            { name: "Jim", type: "Cat" },
          ],
        },
        {
          name: "Samantha",
          gender: "Female",
          age: 40,
          pets: [{ name: "Tabby", type: "Cat" }],
        },
        {
          name: "Alice",
          gender: "Female",
          age: 64,
          pets: [
            { name: "Simba", type: "Cat" },
            { name: "Nemo", type: "Fish" },
          ],
        },
      ],
      gender: "Male",
    };
    const { asFragment } = renderComponent(props);
    expect(asFragment()).toMatchSnapshot();
    const catsList = screen.getByTestId('cats-list')
  expect(catsList).toBeInTheDocument();
  });

  it("it should render the cat list based on gender Male", () => {
    const defaultProps = {
      userObject: [
        {
          name: "Bob",
          gender: "Male",
          age: 23,
          pets: [
            { name: "Garfield", type: "Cat" },
            { name: "Fido", type: "Dog" },
          ],
        },
        {
          name: "Jennifer",
          gender: "Female",
          age: 18,
          pets: [{ name: "Garfield", type: "Cat" }],
        },
        { name: "Steve", gender: "Male", age: 45, pets: null },
        {
          name: "Fred",
          gender: "Male",
          age: 40,
          pets: [
            { name: "Tom", type: "Cat" },
            { name: "Max", type: "Cat" },
            { name: "Sam", type: "Dog" },
            { name: "Jim", type: "Cat" },
          ],
        },
        {
          name: "Samantha",
          gender: "Female",
          age: 40,
          pets: [{ name: "Tabby", type: "Cat" }],
        },
        {
          name: "Alice",
          gender: "Female",
          age: 64,
          pets: [
            { name: "Simba", type: "Cat" },
            { name: "Nemo", type: "Fish" },
          ],
        },
      ],
      gender: "Male",
    };

    const { asFragment } = renderComponent(defaultProps);
    expect(asFragment()).toMatchSnapshot();
  });

  it("it should render the cat list based on gender Female", () => {
    const defaultProps = {
      userObject: [
        {
          name: "Bob",
          gender: "Male",
          age: 23,
          pets: [
            { name: "Garfield", type: "Cat" },
            { name: "Fido", type: "Dog" },
          ],
        },
        {
          name: "Jennifer",
          gender: "Female",
          age: 18,
          pets: [{ name: "Garfield", type: "Cat" }],
        },
        { name: "Steve", gender: "Male", age: 45, pets: null },
        {
          name: "Fred",
          gender: "Male",
          age: 40,
          pets: [
            { name: "Tom", type: "Cat" },
            { name: "Max", type: "Cat" },
            { name: "Sam", type: "Dog" },
            { name: "Jim", type: "Cat" },
          ],
        },
        {
          name: "Samantha",
          gender: "Female",
          age: 40,
          pets: [{ name: "Tabby", type: "Cat" }],
        },
        {
          name: "Alice",
          gender: "Female",
          age: 64,
          pets: [
            { name: "Simba", type: "Cat" },
            { name: "Nemo", type: "Fish" },
          ],
        },
      ],
      gender: "Female",
    };

    const { asFragment } = renderComponent(defaultProps);
    expect(asFragment()).toMatchSnapshot();
  });
});
