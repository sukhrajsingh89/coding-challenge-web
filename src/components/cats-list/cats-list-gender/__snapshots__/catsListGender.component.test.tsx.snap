// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatsList it should render the cat list based on gender Female 1`] = `
<DocumentFragment>
  <div
    data-testid="cats-list"
  />
</DocumentFragment>
`;

exports[`<CatsList it should render the cat list based on gender Male 1`] = `
<DocumentFragment>
  <div
    data-testid="cats-list"
  />
</DocumentFragment>
`;

exports[`<CatsList it should render the component 1`] = `
<DocumentFragment>
  <div
    data-testid="cats-list"
  />
</DocumentFragment>
`;
